text/0093-nft-standard.md

```diff
@@ -0,0 +1,260 @@
+    ///
+    /// @param from - The current owner of the NFT.
+    /// @param to - The new owner.
+    /// @param token_id - The NFT ID.
+    /// @param data - Optional payload to pass to a receiving contract.
+    fn safe_transfer_from(&mut self, from: AccountHash, to: AccountHash, token_id: U256, data: Option<Vec<u8>>);
+
+    /// Transfers the ownership of an NFT without checking the recipient contract.
+    ///
+    /// @param from - The current owner of the NFT.
+    /// @param to - The new owner.
+    /// @param token_id - The NFT ID.
+    fn transfer_from(&mut self, from: AccountHash, to: AccountHash, token_id: U256);
+
+    /// Approves another account to transfer a specific NFT.
+    ///
+    /// @param to - The account that will be granted approval.
+    /// @param token_id - The NFT ID.
+    fn approve(&mut self, to: Option<AccountHash>, token_id: U256);
+
+    /// Revokes approval for a specific NFT.
+    ///
+    /// @param token_id - The NFT ID.
+    fn revoke_approval(&mut self, token_id: U256);
+
+    /// Enables operator approval for all of the caller's NFTs.
+    ///
+    /// @param operator - The operator address to be approved.
+    fn approve_for_all(&mut self, operator: AccountHash);
+
+    /// Revokes operator approval for all of the caller's NFTs.
+    ///
+    /// @param operator - The operator address whose approval is to be revoked.
+    fn revoke_approval_for_all(&mut self, operator: AccountHash);
+
+    /// Gets the approved account for a specific NFT.
+    ///
+    /// @param token_id - The NFT ID.
+    /// @return Option<AccountHash> - Approved account if one exists.
+    fn get_approved(&self, token_id: U256) -> Option<AccountHash>;
+
+    /// Checks if an operator is approved to manage all NFTs of the owner.
+    ///
+    /// @param owner - The NFT owner's address.
+    /// @param operator - The operator to check.
+    /// @return bool - True if the operator is approved for all NFTs, false otherwise.
+    fn is_approved_for_all(&self, owner: AccountHash, operator: AccountHash) -> bool;
+
+    /// Emits NFT-related events.
+    fn emit_event(&self, event: CEP95Event);
+}
+```
+
+## Metadata Extension
+
+[metadata-extension]: #metadata-extension
+
+The metadata extension is OPTIONAL for Casper NFT smart contracts. This extension allows your smart contract to be interrogated for its collection name, symbol, and a URI pointing to metadata describing the assets represented by your NFTs.
+
+### Metadata Interface: CEP95Metadata
+
+```rust
+pub trait CEP95Metadata {
+    /// Returns a descriptive name for a collection of NFTs.
+    fn name(&self) -> String;
+
+    /// Returns a short symbol or abbreviation for the NFT collection.
+    fn symbol(&self) -> String;
+
+    /// Returns a URI pointing to metadata for a specific token.
+    ///
+    /// @param token_id - The NFT ID.
+    /// @return Option<String> - A URI pointing to JSON metadata if available.
+    fn token_uri(&self, token_id: U256) -> Option<String>;
```

Comment:
Maybe let's have a `get_token_metadata` method that return the on-chain metadata and have a separate standard for the metadata structure with the optional following fields:
- `name`
- `description`
- `asset_uri`
- `token_uri`